src/metalookup/app/api.py

```diff
@@ -73,6 +73,51 @@ async def extract(input: Input, request: Request, response: Response, extra: boo
     return result
 
 
+@app.post(
+    "/lrmi-suggestions",
+    response_model=LRMISuggestions,
+    description="Use the available extractors to deduce PoC level suggestions for four of the meta data fields of the LRMI.",
+)
+@cache(
+    expire=24 * 60 * 60 * 28,
+    # Note, we simply modify the cache key here to use the same codebase as for the extract endpoint.
+    key=lambda input, request, response, extra: f"{input.url}-lrmi" if input.splash_response is None else None,
+    backend=cache_backend,
+)
+# request and response arguments are needed for the cache wrapper.
+async def suggest(input: Input, request: Request, response: Response):
+    logger.info(f"Received request for {input.url}")
+
+    result = await manager.extract(input, extra=True)
+
+    def combine(*fields: str) -> Union[MetadataTags, Error]:
+        """
+        Combines the extractors' MetaDataTags for given extractors (the field names used in the Output model).
+        Star rating will be deduced as minimum star rating.
+        If any extractor failed, the combined result is also a failure listing all sub-failures.
+        """
+        fields: Dict[str, Union[Error, MetadataTags]] = {field: getattr(result, field) for field in fields}
```

Comment:
If fields is a string, how can you iterate over it?

Reply:
Ooh you unpack the given list. Mmmh, interesting way, but also quite explizit. I had to get to line 115 to understand line 93 and thus line 99.

Reply:
The function signature `def combine(*fields: str)` defines fields as `tuple[str,...]`. When using `*args` python always assumes that they are of the same type, and you only annotate the type that is in the tuple :)